src/controller/tests/TestServerCommandDispatch.cpp

```diff
@@ -134,6 +134,20 @@ class DispatchTestDataModel : public CodegenDataModelProvider
         static DispatchTestDataModel instance;
         return instance;
     }
+
+    // The Startup method initializes the data model provider with a given context.
+    // This approach ensures that the test relies on a more controlled and explicit data model provider
+    // rather than depending on the code-generated one with undefined modifications.
+    CHIP_ERROR Startup(DataModel::InteractionModelContext context) override
+    {
+        ReturnErrorOnFailure(CodegenDataModelProvider::Startup(context));
+        return CHIP_NO_ERROR;
+    }
+
+protected:
+    // Since the current unit tests do not involve any cluster implementations, we override InitDataModel
```

Comment:
The tests should be implementing their own provider, then, not using a "codegen, but oops, no actual codegen" provider.  I'm OK with this being a temporary hack until the tests are fixed to do the right thing, but it needs to be clearly documented and named as such, and we need a commitment to fix the tests.

Reply:
https://github.com/project-chip/connectedhomeip/issues/36837